Jenkinsfile.github

```
@@ -476,6 +485,19 @@ pipeline {
                 }
             }
         }
+        stage('Prune') {
+            when { expression { params.prune } }
```

Would also run this exclusively, and not when as part of a users build.
```suggestion
            when {
                expression { params.prune }
                expression { !params.buildAndPublish }
            }
```

---

Then to each stage add this to prevent them from running anything when a prune is running.
```groovy
            when {
                expression { !params.prune }
            }
```

So add it to:
- `setup`
- `base layer(s)`
- `common layer(s)`
- `ncn node layers`

---

Basically need to make this pipeline not run anything else except the prune stage when pruning.
As well as never run the prune stage when not pruning.

---

Also needs to be on the `post('Test Results')` step as well, right?

---

Yup
https://github.com/Cray-HPE/node-images/pull/280#discussion_r978774540